packages/server-wallet/src/handlers/update-channel.ts

```diff
@@ -3,32 +3,37 @@ import {SignedStateWithHash, StateVariables, Outcome} from '@statechannels/walle
 import {pipe} from 'fp-ts/lib/function';
 import {ChannelId} from '@statechannels/client-api-schema';
 
-import {SignState} from '../protocols/actions';
+import {SignState, signState} from '../protocols/actions';
 import {ChannelState} from '../protocols/state';
 
 type ChannelStateWithSupported = ChannelState & {
   supported: SignedStateWithHash;
   latestSignedByMe: SignedStateWithHash;
 };
 
-type HandlerResult = Either<UpdateChannelError, SignState>;
 type StepResult = Either<UpdateChannelError, ChannelStateWithSupported>;
+type UpdateChannelResult = Either<UpdateChannelError, SignState>;
+
 export interface UpdateChannelHandlerParams {
   channelId: ChannelId;
   outcome: Outcome;
   appData: string;
 }
 
-export enum Errors {
-  channelNotFound = 'channel not found',
-  invalidLatestState = 'must have latest state',
-  notInRunningStage = 'channel must be in running state',
-  notMyTurn = 'it is not my turn',
-}
-
 export class UpdateChannelError extends Error {
   readonly type = 'UpdateChannelError';
-  constructor(reason: Errors, public readonly data: any = undefined) {
+
+  static readonly reasons = {
+    channelNotFound: 'channel not found',
+    invalidLatestState: 'must have latest state',
+    notInRunningStage: 'channel must be in running state',
+    notMyTurn: 'it is not my turn',
+  } as const;
+
+  constructor(
+    reason: typeof UpdateChannelError.reasons[keyof typeof UpdateChannelError.reasons],
```

Comment:
FYI I played around with making an error-class-factory for about 5 or 10 minutes, and then gave up.

If we use this pattern often, then it would be nice to define `type Values<E> = E[keyof E];`.
then, this can be
```suggestion
    reason: Values<typeof UpdateChannelError.reasons>,
```

Reply:
That does look cleaner: 81cfd0442